### PR TITLE
[Networking][1/2] Added routes for networking

### PR DESCRIPTION
**Pull request by lucyxubroad**
## Overview
- Added FutureNova and infrastructure for implementing networking

## Changes Made
- Added FutureNova
- Created models for data
- Added logic to be able to utilize network routes

## Test Coverage
- Manual testing

## Next Steps (delete if not applicable)
- Replace all data with backend values

## Related PRs or Issues (delete if not applicable)
#27 

</details>

### CODE REVIEW

File: CoffeeChat/Controller/EditDemographicsViewController.swift [hunk 1]
```diff
@@ -227,7 +227,7 @@ class EditDemographicsViewController: UIViewController {
 
 extension EditDemographicsViewController: OnboardingDropdownViewDelegate {
 
-    func updateSelectedFields(tag: Int, isSelected: Bool) {
+    func updateSelectedFields(tag: Int, isSelected: Bool, valueSelected: String) {
         fieldsEntered[tag] = isSelected
```
Comment (camruss) [inline CoffeeChat/Controller/EditDemographicsViewController.swift:231]: Unless this is all the function is intended to do, could you add a `TODO` or fill it out? I imagine it's pretty similar to the one in `DemographicsViewController`.

File: CoffeeChat/Controller/Endpoints.swift [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        return Endpoint(path: "/general/hello/")
+    }
+
+    /// [POST] Authenticate ID token from Google and creates a user if account does not exist
+    static func createUser(idToken: String) -> Endpoint {
+        let body = UserSessionBody(idToken: idToken)
+        return Endpoint(path: "/auth/login/", body: body)
+    }
+
+    /// [POST] Authenticate ID token from Google and creates a user if account does not exist
```
Comment (camruss) [inline CoffeeChat/Controller/Endpoints.swift:39]: Creates or updates as the function name implies?

File: CoffeeChat/Controller/Endpoints.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    static func updateUser(clubs: [String],
+                           graduationYear: String,
+                           hometown: String,
+                           interests: [String],
+                           major: String,
+                           pronouns: String) -> Endpoint {
+        let body = UserUpdateBody(clubs: clubs,
+                            graduationYear: graduationYear,
+                            hometown: hometown,
```
Comment (camruss) [inline CoffeeChat/Controller/Endpoints.swift:48]: Fix the alignment on this so everything is flush

File: CoffeeChat/Models/User.swift [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    let clubs: [String]
+    let firstName: String
+    let googleID: String
+    let graduationYear: String
+    let hometown: String
+    let interests: [String]
+    let lastName: String
+    let netID: String
+    let major: String
```
Comment (camruss) [inline CoffeeChat/Models/User.swift:21]: ... j k l m n o p ...

**Review by camruss — APPROVED** [review]

File: CoffeeChat/Controller/LoginViewController.swift [hunk 5]
```diff
@@ -102,6 +102,17 @@ class LoginViewController: UIViewController {
             self.errorMessageAlertView.transform = .identity
         })
     }
+
+    private func createUser(idToken: String) {
+        NetworkManager.shared.createUser(idToken: idToken).observe { result in
+            switch result {
+            case .value(let response):
+                print(response)
```
Comment (ywsang) [inline CoffeeChat/Controller/LoginViewController.swift:110]: is this just a "placeholder" action for when networking is actually connected? if so, can be add a `TODO`, and if not & it's just for logging purposes, can we do a more formatted/detailed log?

File: CoffeeChat/Controller/Response.swift [hunk 6]
```diff
@@ -0,0 +1,17 @@
+//
```
Comment (ywsang) [inline CoffeeChat/Controller/Response.swift:1]: move this to the `Models` folder?

File: CoffeeChat/Models/Body.swift [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    let interests: [String]
+    let major: String
+    let pronouns: String
+
+}
+
+struct MatchingBody: Codable {
+
+    let netIDs: [String]
+    let schedule: [DaySchedule]
```
Comment (ywsang) [inline CoffeeChat/Models/Body.swift:33]: maybe a comment on whose schedule this is? i.e. does it belong to one of the users, is a merging of both of the users, is all of the overlaps between the users, etc.

File: CoffeeChat/Models/Matching.swift [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+//  Created by Lucy Xu on 5/7/20.
+//  Copyright © 2020 cuappdev. All rights reserved.
+//
+
+import Foundation
+
+struct Matching: Codable {
+
+    let active: Bool
+    let schedule: [DaySchedule]
```
Comment (ywsang) [inline CoffeeChat/Models/Matching.swift:14]: ditto here

**Review by ywsang — COMMENTED** [review]
lgtm! just a couple nitpicky comments

**Comment by ywsang** [comment]
> lgtm! just a couple nitpicky comments

i accidentally just commented my review instead of approving lmao, but i approve

**Review by pnor — APPROVED** [review]
Looks awesome! Excited to see networking 🛰

File: CoffeeChat/Models/User.swift (hunk 4 above)
Comment (lucyxubroad) [reply CoffeeChat/Models/User.swift:21]: lol thanks for the reminder on the alphabet

**Pull request merged**